cdap-ui/app/cdap/components/Market/store/market-store.js

```diff
@@ -21,8 +21,13 @@ import Version from 'services/VersionRange/Version';
 import VersionStore from 'services/VersionStore';
 import isNil from 'lodash/isNil';
 
+function getDefaultMarketHost() {
+  return window.CDAP_CONFIG.marketUrls.length > 0 ? window.CDAP_CONFIG.marketUrls[0] : '';
```

Comment:
This should also be the `window.CDAP_CONFIG.marketBaseUrl`. We cannot assume the market `0` index will be the default.

Reply:
One solution would be to check if there is list of markets passed on and find the host that we host (marketBaseUrl). If there is no list then we should fallback to `marketBaseUrl` which will always be passed.

This assumes that we always set `marketBaseUrl` to be cdap default market.

Reply:
This prompted, we should have a fallback to handle when user removes all market urls. We should the very least say no markets specified. We don't have to address it in this JIRA but we can come back to it later.